### ISSUE TITLE
Any path that let gradient objects access its forward correspondent?

### ISSUE THREAD

**Issue by CaoZhongZ** [facebookarchive/caffe2#1519]
As title, in C++, can a OpGradient object access its corresponding Op object? @Yangqing

**Comment by slayton58** [comment]
Not that I'm aware of. It begs the question of why you want to do this though - everything that needs to be shared between forward and backward ops can be done via input/output blobs.

**Comment by CaoZhongZ** [comment]
OK, can I arbitrarily add one output and let the gradient receive it? Transparent to interface

**Comment by slayton58** [comment]
Yes. You'd just need to specify in the `GradientMaker` for the op that the additional output from the forward pass op needed to be passed as an input to the gradient op.

**Comment by CaoZhongZ** [comment]
Thanks, that's very helpful.

**Comment by slayton58** [comment]
It's not the easiest code to read, but see the [cuDNN RNN op](https://github.com/caffe2/caffe2/blob/master/caffe2/operators/recurrent_op_cudnn.cc#L590-L615) -- the `RNN_SCRATCH` blob is used to share some state between fprop and bprop passes
